Security Policies for GKE backup resources

Tikster123 opened the pull request:
Changes Made-

- Created 10 policies for `google_gke_backup_backup_plan` resource
- Added compliant and non-compliant Terraform test configurations for each policy
- All policies tested and validated with OPA evaluation engine

Resources Covered-

### google_gke_backup_backup_plan
- location - Regional compliance
- deactivated - Backup activation status
- include_secrets - Secret backup enforcement
- include_volume_data - Volume data backup requirement
- backup_schedule - Automated backup scheduling
- retention_days - Retention period compliance
- encryption_key - Customer-managed encryption
- selected_applications - Application selection control
- paused - Schedule pause prevention
- labels - Resource labeling requirements

## Testing
✅ All policies tested with Terraform plan generation
✅ OPA evaluation successful for all policies
✅ Both compliant and non-compliant scenarios validated

Tikster123 committed:
Complete GKE backup policies with working OPA evaluations

Tikster123 committed:
Add location policy for gke_backup_backup_plan_iam_policy

Tikster123 committed:
Merge branch 'gcp/service/gke_backup' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/service/gke_backup

Tikster123 committed:
Add 19 security policies for GKE Backup resources

- Added 9 policies for google_gke_backup_backup_plan_iam_policy
  * location, project, name, roles, members
  * service_accounts, binding_count, cross_project, groups

- Added 10 policies for google_gke_backup_backup_plan
  * location, deactivated, include_secrets, include_volume_data
  * backup_schedule, retention_days, encryption_key
  * selected_applications, paused, labels

- All policies tested with compliant/non-compliant Terraform configurations
- Validated with OPA evaluation engine

Tikster123 committed:
Merge branch 'gcp/service/gke_backup' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/service/gke_backup
git commit -m "Add 19 security policies for GKE Backup resources

- Added 9 policies for google_gke_backup_backup_plan_iam_policy
  * location, project, name, roles, members
  * service_accounts, binding_count, cross_project, groups

- Added 10 policies for google_gke_backup_backup_plan
  * location, deactivated, include_secrets, include_volume_data
  * backup_schedule, retention_days, encryption_key
  * selected_applications, paused, labels

- All policies tested with compliant/non-compliant Terraform configurations
- Validated with OPA evaluation engine"

LegendaryMercury reviewed (commented):
Looks good, You said everything works so ill trust that they do. Ill just get a second opinion from my team before i approve anything though. Just to be sure!

LegendaryMercury commented:
Hey, sorry i have just been told of an issue that i myself have to fix in my code too.

Names have to be c or nc for id in vars.rego c.tf and nc.tf files, numbers like c1 or nc2 are ok.

EG:

<img width="542" height="281" alt="Screenshot 2025-09-06 123216" src="https://github.com/user-attachments/assets/d0017414-9e48-4939-bca1-7e57b0917155" />

Has to be:

<img width="524" height="270" alt="Screenshot 2025-09-06 123243" src="https://github.com/user-attachments/assets/f0208f0e-68c0-4897-a2b6-2238cfecde25" />

See how engine ID has changed to just c (or nc for nc). That has to be done on all c and nc files.
Thanks, sorry for any confusion.

Tikster123 commented:
No worries i understand ill fix it right away

Tikster123 committed:
Resolve merge conflicts

Tikster123 committed:
Restore IAM policies with correct naming convention

LegendaryMercury commented:
You need to change the IDs for each resource in c and nc. Look at Engine ID in my example. See how its just "c", not "chat_engine_id", in the second. That is what needs to be changed.
(I had to change this as well in my code)

Tikster123 committed:
Merge remote-tracking branch 'origin/dev' into gcp/service/gke_backup

Pat-Stuart commented on policies/gcp/gke_backup/backup_plan/labels/policy.rego:
This is not security related and can be removed

Pat-Stuart commented on policies/gcp/gke_backup/backup_plan/location/policy.rego:
Any particular reason for these regions? Why not Aus?

Tikster123 replied:
The regions were selected as examples of commonly used GCP regions for demonstration purposes.

LegendaryMercury commented:
So the naming for the c and nc files isnt right yet. Have a closer look at the example.

It should be the IDs that are c and nc, not just the files names.

Tikster123 committed:
Add 19 security policies for GKE Backup resources

- Added 9 policies for google_gke_backup_backup_plan_iam_policy
  * location, project, name, roles, members
  * service_accounts, binding_count, cross_project, groups

- Added 10 policies for google_gke_backup_backup_plan
  * location, deactivated, include_secrets, include_volume_data
  * backup_schedule, retention_days, encryption_key
  * selected_applications, paused, labels

- All policies tested with compliant/non-compliant Terraform configurations
- Validated with OPA evaluation engine

Tikster123 committed:
Resolve merge conflicts

Tikster123 committed:
Add GKE backup retention_days policy

Tikster123 committed:
Convert large files to Git LFS

Tikster123 committed:
Remove .terraform directories and add to .gitignore

Tikster123 committed:
Implement 8 GKE backup security policies with descriptive naming

Security Policies Implemented:
- australian_region_restriction: Enforce data sovereignty in AU regions
- minimum_retention_period_enforcement: Require 30+ day retention for compliance
- persistent_volume_backup_required: Ensure complete data protection
- kubernetes_secrets_backup_enforcement: Mandate secrets backup coverage
- comprehensive_namespace_coverage: Require all-namespace backup scope
- customer_managed_encryption_required: Enforce CMEK for enhanced security
- mandatory_backup_scheduling: Require automated backup scheduling
- production_cluster_naming_enforcement: Enforce naming conventions
- prevent_backup_plan_deactivation: Prevent accidental backup disabling

Changes Made:
- Removed description and paused policies per review feedback
- Fixed policy value checks for null/empty validation
- Renamed all policy folders to descriptive security-focused names
- Updated package declarations and import paths
- Created comprehensive test configurations for each policy
- Verified all policies work correctly with OPA evaluation

Tikster123 committed:
Merge branch 'gcp/service/gke_backup' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/service/gke_backup
Security Policies Implemented:
- australian_region_restriction: Enforce data sovereignty in AU regions
- minimum_retention_period_enforcement: Require 30+ day retention for compliance
- persistent_volume_backup_required: Ensure complete data protection
- kubernetes_secrets_backup_enforcement: Mandate secrets backup coverage
- comprehensive_namespace_coverage: Require all-namespace backup scope
- customer_managed_encryption_required: Enforce CMEK for enhanced security
- mandatory_backup_scheduling: Require automated backup scheduling
- production_cluster_naming_enforcement: Enforce naming conventions
- prevent_backup_plan_deactivation: Prevent accidental backup disabling

Changes Made:
- Removed description and paused policies per review feedback
- Fixed policy value checks for null/empty validation
- Renamed all policy folders to descriptive security-focused names
- Updated package declarations and import paths
- Created comprehensive test configurations for each policy
- Verified all policies work correctly with OPA evaluation"

Tikster123 committed:
Address review feedback: Improved GKE backup security policies

- Removed description and paused policies per reviewer request
- Fixed policy value checks for null validation (backup_schedule, encryption_key)
- Renamed all policy folders to descriptive security-focused names
- Updated cluster_pattern policy to match actual test configurations
- Created comprehensive test scenarios for all 8 remaining policies
- Verified all policies work correctly with OPA evaluation

Policies Included:
1. australian_region_restriction
2. minimum_retention_period_enforcement
3. persistent_volume_backup_required
4. kubernetes_secrets_backup_enforcement
5. comprehensive_namespace_coverage
6. customer_managed_encryption_required
7. mandatory_backup_scheduling
8. production_cluster_naming_enforcement
9. prevent_backup_plan_deactivation

Tikster123 committed:
Merge branch 'gcp/service/gke_backup' of https://github.com/Hardhat-Enterprises/Policy-Deployment-Engine into gcp/service/vpc_access1_clean

Tikster123 committed:
Resolved merge conflicts and fixed test expectations

Tikster123 committed:
feat: Add Serverless VPC Access Connector security policies

- Added 10 security policies for google_vpc_access_connector
- Region compliance for data sovereignty
- Network validation for production VPCs
- IP CIDR range compliance (RFC 1918)
- Throughput requirements (min/max)
- Instance limits for HA and cost control
- Machine type validation
- Subnet specification checks
- Project restrictions
- Test configurations for each policy

Tikster123 committed:
Add comprehensive GKE backup plan policies

- Implement 9 GKE backup plan policies with proper Rego syntax
- Add australian_region_restriction policy for data sovereignty
- Add comprehensive_namespace_coverage for complete workload protection
- Add customer_managed_encryption_required for enhanced security
- Add kubernetes_secrets_backup_enforcement for security coverage
- Add mandatory_backup_scheduling for automated protection
- Add minimum_retention_period_enforcement for compliance
- Add persistent_volume_backup_required for complete data protection
- Add prevent_backup_plan_deactivation for continuous protection
- Add production_cluster_naming_enforcement for environment identification
- All policies tested and working with helpers.rego framework
- Standard Terraform configurations created for Australian regions

Tikster123 committed:
Add comprehensive GKE backup plan policies

- Implement 9 GKE backup plan policies with proper Rego syntax
- Add australian_region_restriction policy for data sovereignty
- Add comprehensive_namespace_coverage for complete workload protection
- Add customer_managed_encryption_required for enhanced security
- Add kubernetes_secrets_backup_enforcement for security coverage
- Add mandatory_backup_scheduling for automated protection
- Add minimum_retention_period_enforcement for compliance
- Add persistent_volume_backup_required for complete data protection
- Add prevent_backup_plan_deactivation for continuous protection
- Add production_cluster_naming_enforcement for environment identification
- All policies tested and working with helpers.rego framework
- Standard Terraform configurations created for Australian regions

Tikster123 committed:
Add comprehensive GKE backup plan policies

- Implement 9 GKE backup plan policies with proper Rego syntax
- Add australian_region_restriction policy for data sovereignty
- Add comprehensive_namespace_coverage for complete workload protection
- Add customer_managed_encryption_required for enhanced security
- Add kubernetes_secrets_backup_enforcement for security coverage
- Add mandatory_backup_scheduling for automated protection
- Add minimum_retention_period_enforcement for compliance
- Add persistent_volume_backup_required for complete data protection
- Add prevent_backup_plan_deactivation for continuous protection
- Add production_cluster_naming_enforcement for environment identification
- All policies tested and working with helpers.rego framework
- Standard Terraform configurations created for Australian regions

Tikster123 committed:
Add comprehensive GKE backup plan policies

- Implement 9 GKE backup plan policies with proper Rego syntax
- Add australian_region_restriction policy for data sovereignty
- Add comprehensive_namespace_coverage for complete workload protection
- Add customer_managed_encryption_required for enhanced security
- Add kubernetes_secrets_backup_enforcement for security coverage
- Add mandatory_backup_scheduling for automated protection
- Add minimum_retention_period_enforcement for compliance
- Add persistent_volume_backup_required for complete data protection
- Add prevent_backup_plan_deactivation for continuous protection
- Add production_cluster_naming_enforcement for environment identification
- All policies tested and working with helpers.rego framework
- Standard Terraform configurations created for Australian regions

Tikster123 committed:
Add complete gke_backup service with backup_plan, restore_plan, and backup_channel resources and policies

Tikster123 committed:
Add comprehensive GKE backup plan policies

- Implement 9 GKE backup plan policies with proper Rego syntax
- Add australian_region_restriction policy for data sovereignty
- Add comprehensive_namespace_coverage for complete workload protection
- Add customer_managed_encryption_required for enhanced security
- Add kubernetes_secrets_backup_enforcement for security coverage
- Add mandatory_backup_scheduling for automated protection
- Add minimum_retention_period_enforcement for compliance
- Add persistent_volume_backup_required for complete data protection
- Add prevent_backup_plan_deactivation for continuous protection
- Add production_cluster_naming_enforcement for environment identification
- All policies tested and working with helpers.rego framework
- Standard Terraform configurations created for Australian regions

Tikster123 committed:
Add comprehensive GKE backup plan policies

- Implement 9 GKE backup plan policies with proper Rego syntax
- Add australian_region_restriction policy for data sovereignty
- Add comprehensive_namespace_coverage for complete workload protection
- Add customer_managed_encryption_required for enhanced security
- Add kubernetes_secrets_backup_enforcement for security coverage
- Add mandatory_backup_scheduling for automated protection
- Add minimum_retention_period_enforcement for compliance
- Add persistent_volume_backup_required for complete data protection
- Add prevent_backup_plan_deactivation for continuous protection
- Add production_cluster_naming_enforcement for environment identification
- All policies tested and working with helpers.rego framework
- Standard Terraform configurations created for Australian regions
